### PR TITLE
Transform special traits to Mixpanel reserved properties

### PR DESCRIPTION
**Pull request by zihejia**
This PR adds transformation of special traits including 'created', 'email', 'firstName', 'lastName', 'name', 'username', and 'phone' to Mixpanel reserved properties for compatibility with Mixpanel classic integration(https://segment.com/docs/connections/destinations/catalog/mixpanel/#group-using-device-mode). 

This PR also removed the misplaced `segment_source_name` property for the `$set` call. 

## Testing

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/mixpanel/identifyUser/__tests__/index.test.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ describe('Mixpanel.identifyUser', () => {
           $token: MIXPANEL_PROJECT_TOKEN,
           $distinct_id: 'user1234',
           $set: {
-            abc: '123'
+            abc: '123',
+            $name: 'Joe' // transformed to reserved property $name
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/mixpanel/identifyUser/__tests__/index.test.ts:51]:
```suggestion
            $name: 'Joe'
```

File: packages/destination-actions/src/destinations/mixpanel/groupIdentifyUser/__tests__/index.test.ts [hunk 2]
```diff
@@ -39,7 +39,8 @@ describe('Mixpanel.groupIdentifyUser', () => {
           $group_id: 'Mixpanel',
           $set: {
             hello: 'world',
-            company: 'Mixpanel'
+            company: 'Mixpanel',
+            $name: 'test' // validate name -> $name
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/mixpanel/groupIdentifyUser/__tests__/index.test.ts:43]:
```suggestion
            $name: 'test'
```

**Review by marinhero — APPROVED** [review]
Code looks good. Thanks for using our provided libraries! I left a couple of comments since I believe the code comments are a bit redundant since the tests are very succint. Up to you to remove them or not. Consider this inside of the queue for tomorrow's release. Well done!

File: packages/destination-actions/src/destinations/mixpanel/identifyUser/index.ts [hunk 3]
```diff
@@ -66,8 +66,15 @@ const action: ActionDefinition<Settings, Payload> = {
 
     if (payload.traits && Object.keys(payload.traits).length > 0) {
       const traits = {
-        ...omit(payload.traits, ['name']),
-        $name: payload.traits.name  // transform to Mixpanel reserved property
+        ...omit(payload.traits, ['created', 'email', 'firstName', 'lastName', 'name', 'username', 'phone']),
+        // to fit the Mixpanel expections, transform the special traits to Mixpanel reserved property
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/mixpanel/identifyUser/index.ts:70]: Did you mean `expectations` instead of  `expections`?

**Review by marinhero — APPROVED** [review]

File: packages/destination-actions/src/destinations/mixpanel/identifyUser/index.ts (hunk 3 above)
Comment (zihejia) [reply packages/destination-actions/src/destinations/mixpanel/identifyUser/index.ts:70]: oops.

**Pull request merged**